app/inbox/views.py

```diff
@@ -0,0 +1,121 @@
+        all_notifications = []
+        for i in all_pages.page(page):
+            new_notif = i.to_standard_dict()
+            new_notif['username'] = i.from_user.username
+            all_notifications.append(new_notif)
+        params['data'] = all_notifications
+        params['has_next'] = all_pages.page(page).has_next()
+        params['count'] = all_pages.count
+        params['num_pages'] = all_pages.num_pages
+    else:
+        params['num_pages'] = all_pages.num_pages
+    return JsonResponse(params, status=200, safe=False)
+
+
+@login_required
+@require_http_methods(['DELETE'])
+@csrf_exempt
+def delete_notifications(request):
+    """For deleting a notification."""
+    params = {'success': []}
+    try:
+        req_body = json.loads(request.body.decode('utf-8'))
+    except:
+        pass
+    if 'delete' in req_body:
+        for i in req_body['delete']:
+            try:
+                obj = Notification.objects.get(id=i)
+                if obj.to_user.id == request.user.id:
+                    obj.delete()
+                    params['success'].append(True)
+                else:
+                    params['success'].append(False)
+            except Notification.DoesNotExist:
+                params['success'].append(False)
+    return JsonResponse(params, status=200)
+
+
+@login_required
+@require_http_methods(['PUT'])
+@csrf_exempt
+def unread_notifications(request):
+    """Mark a notification as unread."""
+    params = {'success': []}
+    try:
+        req_body = json.loads(request.body.decode('utf-8'))
+    except:
+        pass
+    if 'unread' in req_body:
+        for i in req_body['unread']:
+            try:
+                obj = Notification.objects.get(id=i)
+                if obj.to_user.id == request.user.id:
+                    obj.is_read = False
+                    obj.save()
+                    params['success'].append(True)
+                else:
+                    params['success'].append(False)
+            except Notification.DoesNotExist:
+                params['success'].append(False)
+    return JsonResponse(params, status=200)
+
+
+@login_required
+@require_http_methods(['PUT'])
+@csrf_exempt
+def read_notifications(request):
+    """Mark a notification as read."""
+    params = {'success': []}
+    try:
+        req_body = json.loads(request.body.decode('utf-8'))
+    except:
+        pass
+    if 'read' in req_body:
+        for i in req_body['read']:
+            try:
+                obj = Notification.objects.get(id=i)
+                if obj.to_user.id == request.user.id:
+                    obj.is_read = True
+                    obj.save()
+                    params['success'].append(True)
+                else:
+                    params['success'].append(False)
+            except Notification.DoesNotExist:
+                params['success'].append(False)
+    return JsonResponse(params, status=200)
+
+
+def inbox(request):
+    """Handles the inbox view."""
```

Comment:
More of a nitpick - docstrings should be imperative (this warning will be returned from `pydocstyle`) - `Handle the inbox view.` for example, would resolve the linter warning `D401`